supervisor/dbus/udisks2/block.py

```diff
@@ -0,0 +1,244 @@
+
+    http://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.Block.html
+    """
+
+    def __init__(self, object_path: str) -> None:
+        """Initialize NetworkConnection object."""
+        self.object_path = object_path
+        self.properties = {}
+
+    async def connect(self):
+        """Connect to D-Bus."""
+        self.dbus = await DBus.connect(DBUS_NAME_UDISKS2, self.object_path)
+        await self.update()
+
+    @property
+    @dbus_property
+    def device(self) -> str:
+        """Return device file."""
+        return bytes(self.properties[DBUS_ATTR_DEVICE]).decode()
+
+    @property
+    @dbus_property
+    def id(self) -> str:
+        """Return unique identifer."""
+        return self.properties[DBUS_ATTR_ID]
+
+    @property
+    @dbus_property
+    def read_only(self) -> bool:
+        """Return whether device is read only."""
+        return self.properties[DBUS_ATTR_READ_ONLY]
+
+    @property
+    @dbus_property
+    def symlinks(self) -> list[str]:
+        """Return list of symlinks."""
+        return [
+            bytes(symlink).decode() for symlink in self.properties[DBUS_ATTR_SYMLINKS]
+        ]
+
+    @property
+    @dbus_property
+    def device_number(self) -> int:
+        """Return device number."""
+        return self.properties[DBUS_ATTR_DEVICE_NUMBER]
+
+    @property
+    @dbus_property
+    def configuration(
+        self,
+    ) -> dict[str, list[FstabConfigDetails | CrypttabConfigDetails]]:
+        """Return device configuration."""
+        configuration = defaultdict(list)
+        for type_, details in self.properties[DBUS_ATTR_CONFIGURATION]:
+            configuration[type_].append(
+                FstabConfigDetails.from_dict(details)
+                if "dir" in details
+                else CrypttabConfigDetails.from_dict(details)
+            )
+        return dict(configuration)
+
+    @property
+    @dbus_property
+    def size(self) -> int:
+        """Return size."""
+        return self.properties[DBUS_ATTR_SIZE]
+
+    @dbus_connected
+    async def add_configuration_item(
+        self,
+        type_: str,
+        details: FstabConfigDetails | CrypttabConfigDetails,
+        options: dict[str, Any] = None,
+    ):
+        """Add new configuration item."""
+        if not options:
+            options = {}
+        await self.dbus.Block.AddConfigurationItem(
+            ("(sa{sv})", (type_, details.to_dict())), ("a{sv}", options)
+        )
+        await self.update()
+
+    @dbus_connected
+    async def remove_configuration_item(
+        self,
+        type_: str,
+        details: FstabConfigDetails | CrypttabConfigDetails,
+        options: dict[str, Any] = None,
+    ):
+        """Remove existing configuration item."""
+        if not options:
+            options = {}
+        await self.dbus.Block.RemoveConfigurationItem(
+            ("(sa{sv})", (type_, details.to_dict())), ("a{sv}", options)
+        )
```

Comment:
```suggestion
        await self.dbus.Block.call_remove_configuration_item(
            (type_, details.to_dict()), options
        )
```
Same comment as above, we have to get variants and signatures right in each item in this dictionary, can't just pass normal python dictionaries.

Reply:
Suggested changes are accounted for. Leaving open because we need to adjust the dictionaries to have a `Variant` as the value for each key.